test/unit/test_diagram.py

```diff
@@ -30,14 +38,33 @@ def test_job_data():
     assert job_data.headers() == ['Job Name', 'Queue Time', 'Run Time', 'Status']
     assert job_data.values() == ['', datetime.timedelta(0), datetime.timedelta(0), '']
     assert job_data.number_of_columns() == 4
-
     
 def test_job_agg_data():
-    """
-    function to test the Class JobAggData inside autosubmit/monitor/diagram.py
-    """
+    """ function to test the Class JobAggData inside autosubmit/monitor/diagram.py """
     job_agg = JobAggData()
     assert job_agg.headers() == ['Section', 'Count', 'Queue Sum', 'Avg Queue', 'Run Sum', 'Avg Run']
     assert job_agg.values() == [{}, 0, datetime.timedelta(0), datetime.timedelta(0),
                                 datetime.timedelta(0), datetime.timedelta(0)]
     assert job_agg.number_of_columns() == 6
+
+@patch('autosubmit.monitor.diagram._aggregate_jobs_by_section')
+@patch('autosubmit.monitor.diagram._create_table')
+@patch('autosubmit.monitor.diagram._create_csv')
+def test_create_stats_report(jobs_by_association, create_table, create_csv):
+    """ function to test the function create_stats_report inside autosubmit/monitor/diagram.py """
+    expid = "a000"
+    jobs_data = [
+        Job('test', "a29z", "COMPLETED", 200),
+        Job('test', "a28z", "COMPLETED", 200),
+        Job('test', "a27z", "COMPLETED", 200),
+        Job('test', "a26z", "FAILED", 10)
+    ]
+
+    with tempfile.TemporaryDirectory() as temp_dir:
+        if Path(temp_dir).exists() is False:
+            Path(temp_dir).mkdir(parents=True, mode=0o777)
+
+    jobs_by_association.return_value = [JobData(),JobAggData()]
+    create_status = diagram.create_stats_report(expid, jobs_data, {}, temp_dir,True,True,False,None,
+                                                None,None)
+    assert create_status is False
```

Comment:
:laughing: this is not really fair. This is the function you mentioned that receives a `bool` and returns that same `bool`. So this test will pass whether we break the `create_stats_report` or not (the focus when writing tests must be to prevent bugs, not only coverage).

I think a better test would be one that either produces a plot somewhere in a temp dir, and then we assert the plot was created, has more than 0 bytes, and maybe that's that. Or then mock just the matplotlib part, and confirm the mock was called as expected.